python/tskit/trees.py

```diff
@@ -10945,12 +10945,78 @@ def impute_unknown_mutations_time(
     def ld_matrix(
         self,
         sample_sets=None,
-        sites=None,
-        positions=None,
         mode="site",
         stat="r2",
+        sites=None,
+        positions=None,
         indexes=None,
     ):
+        r"""
+        Returns a matrix of the selected two-locus statistic (default :math:`r^2`)
+        computed from sample allelic states or branch lengths. This method is
+        implemented to run as either an LD matrix representing the LD between all
+        selected sites ("site" mode, producing a site×site matrix) or from the branch
+        lengths of all subtree partitions generated by each node in a tree ("branch"
+        mode, producing a tree×tree matrix).
+
+        In the site mode, the sites under consideration can be restricted using the
+        ``sites`` argument. Sites can be passed as a list of lists, specifying the
+        ``[[row_sites], [col_sites]]`` or by specifying ``[all_sites]``, where a square
```

Comment:
Though note that it doesn't have to be "all sites". It can be a single subset of sites on the tree, in which case we get a square matrix.